packages/compass-indexes/src/modules/search-indexes.ts

```diff
@@ -450,7 +451,20 @@ export const updateIndex = (
   indexDefinition: Document
 ): IndexesThunkAction<Promise<void>> => {
   return async function (dispatch, getState) {
-    const { namespace, dataService } = getState();
+    const {
+      namespace,
+      dataService,
+      searchIndexes: { indexes },
+    } = getState();
+
+    const currentIndexDefinition = indexes.find(
+      (x) => x.name === indexName
+    )?.latestDefinition;
+    if (isEqual(currentIndexDefinition, indexDefinition)) {
+      dispatch(closeUpdateModal());
```

Comment:
I can't immediately see where we close the update modal in the normal case?

Reply:
(Was trying to run this locally and manually test it but now bootstrap is broken for me. unrelated to this PR, though..)

Reply:
do we want to show the error message to the user in this case? I am not sure how to handle this case and the best for me is to dismiss the modal as user clicked submit. Maybe we show toast with text "No changes made to the index" or something?

Reply:
> (Was trying to run this locally and manually test it but now bootstrap is broken for me. unrelated to this PR, though..)

It frequently fails for me since we updated `lerna` (i think so)

Reply:
It was just this again:

```
~/mongo/compass % python3

You have not agreed to the Xcode and Apple SDKs license. You must agree to the license below in order to use Xcode.
Press enter to display the license:
```